src/providers/ProtocolData.jsx

```diff
@@ -15,18 +15,34 @@ const buildContractConfigs = (contract, functionNames, chainId) =>
     chainId,
   }));
 
+const buildContractConfigsWithArgs = (contract, calls, chainId) =>
```

Comment:
So I think we just combine these two contract builder functions so we just have:

```js
const buildContractConfigs = (contract, calls, chainId) =>
  calls.map((call) => ({
    ...contract,
    ...call,
    chainId,
  }));
```

And then update the existing ones that just pass in a function to:

```js
const buildContractConfig = (functionName) => ({ functionName });

   ...buildContractConfigs(
      userManagerContract,
      userManagerFunctionNames.map(buildContractConfig),
      chainId
    )
```

Hopefully that makes sense.